### PR TITLE
Update lndclient, add labels to on-chain transactions

### PR DESCRIPTION
**Pull request by guggero**
We update to the latest lndclient that now requires a label when
publishing an on-chain transaction. Instead of just adding an empty
string, we add proper labels.

#### Pull Request Checklist
- [X] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: labels/labels.go [hunk 1]
```diff
@@ -26,6 +37,20 @@ var (
 	// ErrReservedPrefix is returned when a label contains the prefix
 	// which is reserved for internally produced labels.
 	ErrReservedPrefix = errors.New("label contains reserved prefix")
+
+	// loopOutSweepSuccess is the label used for loop out swaps to sweep
+	// the HTLC in the success case.
+	LoopOutSweepSuccess = fmt.Sprintf(
```
Comment (carlaKC) [inline labels/labels.go:43]: We have two kinds of labels overlapping here - labels for swaps and labels for individual transactions. 
* For swap labels, we use the reserved prefix to ensure that people don't label manual swaps as autoloops (which would mess up our budget). 
* For lnd labels, we do have internal labels (such as channel opens), but do not have a reserved prefix (perhaps we should have, but I didn't think of the requirement at the time, and if people want to intentinally mess up our labelling it's not _really_ our problem, vs autoloop where it could cause actual issues). 

So tldr, I don't think we need the reserved label on these?

File: labels/labels.go [hunk 2]
```diff
@@ -17,6 +17,17 @@ const (
 	// autoOut is the label used for loop out swaps that are automatically
 	// dispatched.
 	autoOut = "autoloop-out"
+
+	// loopOutSweepSuccess is the label used for loop out swaps to sweep
+	// the HTLC in the success case.
+	loopOutSweepSuccess = "loop-out-sweep-success"
```
Comment (carlaKC) [inline labels/labels.go:23]: I'm currently working on custom categories for faraday reports (lightninglabs/faraday/pull/85) which allow you to create a new category in your statement by matching tx labels to a regex. Perhaps add an explicit prefix var (`loopd`) and then make all these labels build on that so it's clear that we need the prefix? Nice to have a consistent format across our services building on lnd.

File: release_notes.md [hunk 3]
```diff
@@ -9,6 +9,13 @@ This file tracks release notes for the loop client.
 * All of the items under the "Next Release" heading should be included in the release notes.
 * As part of the PR that bumps the client version, the "Next Release" heading should be replaced with the release version including the changes.
 
+## Next release
+
+#### NewFeatures
```
Comment (carlaKC) [inline release_notes.md:14]: nit/ask: do you mind copying the `Breaking Changes` and `Bug Fixes` headings in as well? Can see this getting inconsistent over time if we don't copy them across with each new release.

File: labels/labels.go [hunk 4]
```diff
@@ -26,6 +37,20 @@ var (
 	// ErrReservedPrefix is returned when a label contains the prefix
 	// which is reserved for internally produced labels.
 	ErrReservedPrefix = errors.New("label contains reserved prefix")
+
+	// loopOutSweepSuccess is the label used for loop out swaps to sweep
```
Comment (bhandras) [inline labels/labels.go:41]: nit: capitalize

**Review by bhandras — APPROVED** [review]
LGTM, nice! :+1:

**Review by carlaKC — COMMENTED** [review]
Really happy to see this change 🥇 :label_all_da_tings:

Some ponderings on label format from me. 

We also currently just add a memo saying "swap" to loop in invoices, which could probably be more descriptive, but that may break some existing checks/logic so perhaps leave for this PR.

File: labels/labels.go (hunk 1 above)
Comment (guggero) [reply labels/labels.go:43]: Ah, I missed the different purposes of the labels. I moved the `lnd` level labels to its own file and added a more verbose pattern that contains some meta information as well.

File: release_notes.md (hunk 3 above)
Comment (guggero) [reply release_notes.md:14]: Sure thing!

**Review by carlaKC — APPROVED** [review]
magical ✨
